### PR TITLE
[SPARK-36257][SQL][3.2] Updated the version of TimestampNTZ related changes as 3.3.0

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

As we decided to release TimestampNTZ type in Spark 3.3, we should update the versions of TimestampNTZ related changes as 3.3.0.

### Why are the changes needed?

Correct the versions in documentation/code comment.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by gengliangwang** [comment]
This is for branch-3.2. There are some code differences after https://github.com/apache/spark/pull/33444.

**Review by HyukjinKwon — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala [hunk 1]
```diff
@@ -140,7 +140,7 @@ object Encoders {
    * Creates an encoder that serializes instances of the `java.time.Duration` class
    * to the internal representation of nullable Catalyst's DayTimeIntervalType.
    *
-   * @since 3.2.0
+   * @since 3.3.0
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala:143]: Please, revert it back. This is related to interval types.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala (hunk 1 above)
Comment (gengliangwang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala:143]: ok, there is something wrong during cherry pick

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to 3.2. Don't think the failed tests are related to the changes.
Thank you, @gengliangwang and @HyukjinKwon for review.

**Comment by dongjoon-hyun** [comment]
+1, LGTM. Thank you, @gengliangwang and all.

**Pull request closed without merging**